image-rs/src/layer_store.rs

```diff
@@ -0,0 +1,79 @@
+use std::fs;
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+/// The LayerStore abstracts the image layers storage on the file system.
+pub struct LayerStore {
+    /// OCI image layer data store dir.
+    pub data_dir: PathBuf,
+
+    /// Next layer index
+    layers_index: AtomicUsize,
+}
+
+impl LayerStore {
+    /// Retrieves the next layer index by computing the maximum index from the
+    /// existing layers in data dir.
+    pub fn get_layer_index(data_dir: PathBuf) -> AtomicUsize {
+        let mut next = 0;
+        if data_dir.exists() {
```

Comment:
what if data_dir doesn't exist, do we need to error out in this case? if no, and if 0 is a proper value, then we can early exit (and safe a level of indentation):

```rust
if (!data_dir.exists() {
  return AtomicUsize::new(0);
}
...
```